File: modules/client-java/src/main/java/be/vlaanderen/awv/atom/javaapi/FeedProcessingException.java
package be.vlaanderen.awv.atom.javaapi;

import be.vlaanderen.awv.atom.FeedPosition;

public class FeedProcessingException extends RuntimeException {
    public final FeedPosition feedPosition;
    public FeedProcessingException(String message, FeedPosition feedPosition) {
        super(message);
        this.feedPosition = feedPosition;
    }
}

File: modules/client-scala/src/main/scala/be/wegenenverkeer/atom/FeedProcessingException.scala
package be.wegenenverkeer.atom

case class FeedProcessingException(entryIdOpt:Option[String], message:String) extends RuntimeException {

  /**
   * Returns the entry id or null if none is available.
   * (For easy access from Java API)
   */
  def entryId = entryIdOpt.orNull

  override def getMessage: String = message
}
